src/context/walletContext/walletContext.tsx

```diff
@@ -51,166 +43,137 @@ export const WalletProvider = ({ children }: { children: ReactNode }) => {
     ////// User props
     const { addAccount, user } = useAuth()
 
-    ////// Selected Wallet
-    const [selectedWallet, setSelectedWallet] = useState<interfaces.IWallet | undefined>(undefined) // TODO: this is the var that should be exposed for the app to consume, instead of const { address } = useWallet() anywhere
+    const [selectedAddress, setSelectedAddress] = useState<string | undefined>(undefined)
 
-    ////// Wallets
-    const { data: wallets } = useQuery<interfaces.IWallet[]>({
+    const isWalletConnected = useCallback(
+        (wallet: interfaces.IDBWallet): boolean => {
+            if (isPeanut(wallet) && kernelClientAddress) {
+                return isKernelClientReady && areAddressesEqual(kernelClientAddress, wallet.address)
+            } else if (wagmiAddress) {
+                return isWagmiConnected && areAddressesEqual(wagmiAddress, wallet.address)
+            }
+            return false
+        },
+        [isKernelClientReady, kernelClientAddress, isWagmiConnected, wagmiAddress]
+    )
+
+    const createDefaultDBWallet = (address: string): interfaces.IDBWallet => ({
+        walletProviderType: interfaces.WalletProviderType.BYOW,
+        protocolType: interfaces.WalletProtocolType.EVM,
+        address,
+    })
+
+    const { data: wallets = [] } = useQuery<interfaces.IWallet[]>({
         queryKey: ['wallets', user?.accounts, wagmiAddress],
```

Comment:
nit: queryKey should containt all dependencies of isWalletConnected

I added a dev dependency to detect this automatically

Reply:
nvm the linter doesn't seeem to work